github-status/url_builder.go

```diff
@@ -0,0 +1,81 @@
+package function
+
+import (
+	"github.com/openfaas/openfaas-cloud/sdk"
+	"os"
+	"strings"
+)
+
+func buildPrettyURL(url string, success, isStack bool, event *sdk.Event) string {
+	if len(url) == 0 {
+		return ""
+	}
+	if success {
+		if isStack {
+			urlOut := strings.Replace(url, "user", "system", 1)
+			return replaceFunctionSuffix(urlOut, "dashboard") + "/" + event.Owner
+		} else {
+			urlOut := strings.Replace(url, "user", strings.ToLower(event.Owner), 1)
+			return replaceFunctionSuffix(urlOut, event.Service)
+		}
+	}
+	urlOut := strings.Replace(url, "user", "system", 1)
+
+	if isStack {
+		return replaceFunctionSuffix(urlOut, "dashboard") + "/" + event.Owner
+	}
+	return replaceFunctionSuffix(urlOut, "dashboard") + "/" + event.Owner + "/" + event.Service + "/build-log?repoPath=" + event.Owner + "/" + event.Repository + "&commitSHA=" + event.SHA
+}
+
+func buildPublicURL(url, owner, service string, success, isStack bool) string {
+
+	if strings.HasSuffix(url, "/") == false {
+		url = url + "/"
+	}
+
+	if success && !isStack {
+		serviceValue := sdk.FormatServiceName(owner, service)
+		url = url + "function/" + serviceValue
+	} else {
+		url = url + "function/system-dashboard"
+	}
+	return url
+}
+
+func BuildPublicStatusURL(status, statusContext string, event *sdk.Event) string {
```

Comment:
Why is this method now public? 

What does the TODO comment mean?

Reply:
My bad on these, tried to get this in before leaving the office and missed that